Add instructions on GitLab CIViz <> Infra correlation (#14895)
DataDog/documentation: content/en/continuous_integration/setup_pipelines/gitlab.md

**Note**: The Pipelines page shows data for only the default branch of each repository.

### Infrastructure metric correlation

> If you are using self-hosted GitLab runners you can correlate jobs to the infrastructure that it is running them.

Reviewer: ```suggestion
If you are using self-hosted GitLab runners, you can correlate jobs with the infrastructure that is running them.
```

For this feature to work the GitLab runner must have a tag of the form `host:<hostname>`. Tags can be added whilst

> [registering a new runner][6] or through the UI, in **Settings > CI/CD > Runners** and editing the appropiate runner,
> or by manually updating the `config.toml` for existing runners.

Reviewer 2: ```suggestion
[registering a new runner][6], or modifying an already registered one through the Admin UI in **Settings > CI/CD > Runners** or by editing the `config.toml`.
```

Author: I've reword it a bit so it is clearer

> For this feature to work the GitLab runner must have a tag of the form `host:<hostname>`. Tags can be added whilst
> [registering a new runner][6]. For existing runners tags can be added by updating the runner's `config.toml` or
> through the UI by going to **Settings > CI/CD > Runners** and editing the appropiate runner.

Reviewer: ```suggestion
For this feature to work, the GitLab runner must have a tag of the form `host:<hostname>`. Tags can be added while
[registering a new runner][6]. For existing runners, add tags by updating the runner's `config.toml`. Or add tags
through the UI by going to **Settings > CI/CD > Runners** and editing the appropriate runner.
```

> Once setup CI Visibility will automatically detect the host. To see the metrics just click on a job span in the trace

Reviewer 2: ```suggestion
After this steps, CI Visibility adds the hostname to each job. To see the metrics click on a job span in the trace
```

> After this steps, CI Visibility adds the hostname to each job. To see the metrics click on a job span in the trace
> view and in the drawer a new tab named **Infrastructure** will show which contains the host metrics.

Reviewer: ```suggestion
After these steps, CI Visibility adds the hostname to each job. To see the metrics, click on a job span in the trace
view. In the drawer, a new tab named **Infrastructure** appears which contains the host metrics.
```
